presto-main/src/main/java/io/prestosql/server/QuerySessionSupplier.java

```diff
@@ -72,6 +72,11 @@ public Session createSession(QueryId queryId, SessionContext context)
     {
         Identity identity = context.getIdentity();
         accessControl.checkCanSetUser(identity.getPrincipal(), identity.getUser());
+        context.getAuthenticatedIdentity().ifPresent(authenticatedIdentity -> {
```

Comment:
would be good to have a comment why the check is done only `IfPresent`, so i'd not use `IfPresent` at all:

```java
if (! ... isEmpty()) {
   do the auth check
}
else {
  // ....
}
```